examples/codes/external_code.ex.py

```diff
@@ -0,0 +1,387 @@
+# ---
+# jupyter:
+#   jupytext:
+#     cell_metadata_filter: tags,-all
+#     notebook_metadata_filter: -jupytext.text_representation.jupytext_version
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# %% tags=["remove-cell"]
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021-2023 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh,
+#                         J. Morris, D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+"""
+External Codes Example
+"""
+# %%
+from dataclasses import asdict, dataclass
+from enum import auto
+from pprint import pprint
+from typing import Dict, List, Union
+
+from bluemira.base.parameter_frame import Parameter, ParameterFrame
+from bluemira.codes.interface import CodesSetup, CodesSolver, CodesTask, CodesTeardown
+from bluemira.codes.interface import RunMode as BaseRunMode
+from bluemira.codes.params import MappedParameterFrame
+from bluemira.codes.utilities import ParameterMapping
+
+# %% [markdown]
+# # External Code Wrapping
+#
+# This example goes though the minimal steps taken to wrap an external code and
+# retrive its outputs
+#
+# Firstly we define the options available in the code and its name.
+# In this example we're wrapping the python script
+# [ext_code_script.py](ext_code_script.py).
+# Unusually the BINARY global variable is a list instead of a string because there are
+# two base commands to run the script from the commandline.
+#
+# The program has been named "External Code" for simiplicity and is used to help the user
+# trace where variables come from. It is a simple script that slightly modifies
+# a file provided to it.
+#
+# There are 3 dataclasses containing the commandline options for the code,
+# the input parameters and the output parameters.
+
+# %%
+PRG_NAME = "External Code"
+BINARY = ["python", "ext_code_script.py"]
```

Comment:
I'm getting an error when I run the example from the root bluemira directory. Could we get the path to the executable script relative to the example file, or will that mess things up if we're generating the notebook?

Reply:
it should be possible because it will be in the same folder as the script